paths/paths_unix.go

```diff
@@ -62,10 +63,21 @@ func xdgDataHome() string {
 	return filepath.Join(os.Getenv("HOME"), ".local/share")
 }
 
-func ensureStateDirPerms(dirPath string) error {
-	if filepath.Base(dirPath) != "tailscale" {
+func ensureStateDirPerms(dir string) error {
+	if filepath.Base(dir) != "tailscale" {
 		return nil
 	}
-
-	return os.Chmod(dirPath, 0700)
+	fi, err := os.Stat(dir)
+	if err != nil {
+		return err
+	}
+	if !fi.IsDir() {
+		return fmt.Errorf("expected %v to be a directory; is %v", dir, fi.Mode())
```

Comment:
Did you want to perhaps quote the directory name with %q?

Reply:
Done.